morph_tool/resampling.py

```diff
@@ -0,0 +1,201 @@
+"""Resampling for morphio morphologies
+"""
+
+import numpy as np
+import morphio
+
+
+def _vertex_path_lengths(points):
+    """Using the segment length assign a path length for each vertex in the
+    points array.
+
+    Args:
+        segment_lengths (np.ndarray): (N,) array of floats corresponding to
+            segment lengths of a section
+
+    Example:
+        [1, 2, 3] -> [0, 1, 3, 6]
+
+    Notes:
+        The last element in the path length array corresponds to the total
+        length of the segments.
+    """
+    segment_lengths = np.linalg.norm(points[1:] - points[:-1], axis=1)
+
+    path_lengths = np.empty(len(segment_lengths) + 1, dtype=np.float32)
+    path_lengths[0] = 0.
+    path_lengths[1:] = np.cumsum(segment_lengths)
+    return path_lengths
+
+
+def _resample_from_linear_density(points, linear_density):
+    """Creates a resampling of K-2 elements on the polyline defined by
+    N `points`, omitting the first and last values which should remain
+    unchanged. The number of elements is determined by the `linear_density`.
+
+    The new sampling is represented by the line segment parametric equation:
+
+    v'[k] = v[ids[k]] + fractions[k] * (v[ids[k] + 1] - v[ids[k]])
+
+    where (v'[k], v[k]) can be either `points` or any other property mapped
+    on `points`.
+
+    Args:
+        points (np.ndarray): (N, 3) Array of consecutive points definining
+        segments.
+        linear_density (float): Target number of points per micron
+
+    Returns:
+        tuple of numpy arrays:
+            - ids (np.array): (K-2,) Array of positional ids
+            - fractions (np.array): (K-2,) Parametric fractions
+
+    Notes:
+        Given consecutive points [p0, p1, ..., pN] a resampling will create
+        new points on the polyline defined by these points. This means
```

Comment:
It would be good to explain that this function does not create new points, but fractions = weights.